### PR TITLE
Add `statusCode` field for `CommonResponse`

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes [#2879](https://github.com/ballerina-platform/ballerina-standard-library/issues/2879)

#### Publisher Client
- Success cases
```ballerina
websubhub:TopicRegistrationSuccess response = check publisherClientEp->registerTopic("https://sample.topic.com");
int statusCode = response.statusCode;
```
- Failure Cases
```ballerina
websubhub:Acknowledgement|websubhub:UpdateMessageError response = publisherClientEp->notifyUpdate("https://sample.topic.com");
if response is websubhub:UpdateMessageError {
  websubhub:CommonResponse errorDetails = response.detail();
  int statusCode = errorDetails.statusCode;
  // implement logic
}
```

#### Hub Client
- Success cases
```ballerina
websubhub:ContentDistributionSuccess response = check hubClientEp->notifyContentDistribution(msg);
int statusCode = response.statusCode;
```
- Failure Cases
```ballerina
websubhub:ContentDistributionSuccess|error response = hubClientEP->notifyContentDistribution(msg);
if response is ContentDeliveryError {
  websubhub:CommonResponse errorDetails = response.detail();
  int statusCode = errorDetails.statusCode;
  // implement logic
}
```

### Hub Responses
- Developer can override the default response status codes.
```ballerina
service /websubhub on securedListener {
    remote function onRegisterTopic(websubhub:TopicRegistration message, http:Request req)
                                returns websubhub:TopicRegistrationSuccess|websubhub:TopicRegistrationError {
        error? auth = authenticate(req);
        if auth is error {
            return error websubhub:TopicRegistrationError("Auth error occurred", auth, statusCode = http:STATUS_UNAUTHORIZED);
        }
        // implement logic
        return {
          statusCode: http:STATUS_OK,
          body: "Topic registration successful"
         };
    }
    
    // implement other remote methods
}
```
#### Default Status Codes
- Following are the default status codes for `hub` responses.

| Response | Default Status Code |
| ------------- | --------------------------- |
| websubhub:TOPIC_REGISTRATION_SUCCESS | HTTP OK (200) |
| websubhub:TOPIC_REGISTRATION_ERROR | HTTP OK (200) |
| websubhub:TOPIC_DEREGISTRATION_SUCCESS | HTTP OK (200) |
| websubhub:TOPIC_DEREGISTRATION_ERROR | HTTP OK (200) |
| websubhub:ACKNOWLEDGEMENT | HTTP OK (200) |
| websubhub:UPDATE_MESSAGE_ERROR | HTTP OK (200) |
| websubhub:SUBSCRIPTION_ACCEPTED | HTTP ACCEPTED (202 |
| websubhub:BAD_SUBSCRIPTION_ERROR | HTTP BAD REQUEST (400 |
| websubhub:INTERNAL_SUBSCRIPTION_ERROR | HTTP INTERNAL SERVER  ERROR (500) |
| websubhub:UNSUBSCRIPTION_ACCEPTED | HTTP ACCEPTED (202 |
| websubhub:BAD_UNSUBSCRIPTION_ERROR | HTTP BAD REQUEST (400 |
| websubhub:INTERNAL_UNSUBSCRIPTION_ERROR | HTTP INTERNAL SERVER  ERROR (500) |

## Checklist
- [X] Linked to an issue
- [x] Updated the changelog
- [X] Added tests
- [ ] Updated the spec

### CODE REVIEW

File: ballerina/commons.bal [hunk 1]
```diff
@@ -261,56 +279,71 @@ public type SubscriptionTemporaryRedirect record {
...
-public final UpdateMessageError UPDATE_MESSAGE_ERROR = error UpdateMessageError("Error in accessing content");
+public final UpdateMessageError UPDATE_MESSAGE_ERROR = error UpdateMessageError(
+    "Error in accessing content", statusCode = http:STATUS_OK);
 
 # Common response, which could be used for `websubhub:SubscriptionAccepted`.
 public final readonly & SubscriptionAccepted SUBSCRIPTION_ACCEPTED = {};
 
 # Common response, which could be used for `websubhub:BadSubscriptionError`.
-public final BadSubscriptionError BAD_SUBSCRIPTION_ERROR = error BadSubscriptionError("Bad subscription request");
+public final BadSubscriptionError BAD_SUBSCRIPTION_ERROR = error BadSubscriptionError(
```
Comment (chamil321) [inline ballerina/commons.bal:322]: Shall we default the status codes for these errors? I feel like the status code is a bit redundant. If used does not need the default status code, they can override. We can created a specific Error detail for the `BadSubscriptionError` and do following
```ballerina
public type BadSubscriptionErrorDetail record {
    *CommonResponse
    int statusCode = http:STATUS_BAD_REQUEST;
};

public type BadSubscriptionError distinct (Error & error<BadSubscriptionErrorDetail>);
```

Comment (ayeshLK) [reply ballerina/commons.bal:322]: @shafreenAnfar WDYT ?

Comment (chamil321) [reply ballerina/commons.bal:322]: What if they want to change the error message? :) 
In that case they cant use websubhub:BAD_SUBSCRIPTION_ERROR right. However websubhub:BAD_SUBSCRIPTION_ERROR declaration also does not need the status code if you change the error definition as suggested

Comment (chamil321) [reply ballerina/commons.bal:322]: My suggestion is to keep the defaults at the definition level. So both theses error constants and the user created one will have the defaults. With your approach, only constants have defaults but not the definition.

Comment (ayeshLK) [reply ballerina/commons.bal:322]: Understood. Seems a good improvement to be added :thumbsup:

Comment (ayeshLK) [reply ballerina/commons.bal:322]: It seems like there is a lang-level limitation for having intersection types with default values [1]

[1] - [https://github.com/ballerina-platform/ballerina-spec/issues/544#issuecomment-778099662](https://github.com/ballerina-platform/ballerina-spec/issues/544#issuecomment-778099662)

**Pull request merged**